addons/mrp/models/stock_move.py

```diff
@@ -278,6 +278,9 @@ def _action_confirm(self, merge=True, merge_into=False):
         # we go further with the list of ids potentially changed by action_explode
         return super(StockMove, moves)._action_confirm(merge=merge, merge_into=merge_into)
 
+    def _should_explode_move(self, move):
+        return (not move.picking_type_id and not self.env.context.get('is_scrap')) or (move.production_id and move.production_id.product_id == move.product_id)
```

Comment:
The condition is not the same as the initial one
Also, don't forget the `super` call
```suggestion
        can_be_exploded = not move.picking_type_id or (move.production_id and move.production_id.product_id == move.product_id
        return super()._should_explode_move() and can_be_exploded
```